libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx

```diff
@@ -0,0 +1,299 @@
+        }
+      }
+
+      for (const service of toCreateOtherServices) {
+        if (service.title) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: ServiceEnum.Other,
+                  serviceOther: service.title,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with title ${service.title}:`,
+              error
+            );
+          }
+        }
+      }
+
+      refetch();
+      setIsModalVisible(false);
+    } catch (e) {
+      console.error('Error during service submission:', e);
+    }
+  };
+
+  const closeModal = () => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
+    const initialServiceOthers = initialServices
+      .filter((item) => item.service === ServiceEnum.Other)
+      .map((service) => ({
+        id: service.id,
+        title: service.serviceOther || '',
+      }));
+    setIsModalVisible(false);
+    setServiceOthers(initialServiceOthers);
+    setServices(newInitialServices);
+  };
+
+  useEffect(() => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
+    const initialServiceOthers = initialServices
+      .filter((item) => item.service === ServiceEnum.Other)
+      .map((service) => ({
+        id: service.id,
+        title: service.serviceOther || '',
+      }));
+
+    setServiceOthers(initialServiceOthers);
+    setServices(newInitialServices);
+  }, [initialServices]);
+
+  return (
+    <Modal
+      mt={Spacings.sm}
+      vertical
+      closeButton
+      closeModal={closeModal}
+      isModalVisible={isModalVisible}
+    >
+      <KeyboardAwareScrollView
+        style={{
+          flex: 1,
+          backgroundColor: Colors.WHITE,
+        }}
+        keyboardShouldPersistTaps="handled"
+      >
+        <ScrollView
+          contentContainerStyle={{
+            flexGrow: 1,
+            gap: Spacings.sm,
+            paddingBottom: Spacings.md,
+          }}
+          style={{ paddingHorizontal: Spacings.md }}
+        >
+          <TextBold> Services</TextBold>
+          <TextRegular mt="xxs" mb="sm">
+            Select the services to your client in this interaction.
+          </TextRegular>
+          {Services.map((service) => (
+            <View key={service.title}>
+              <TextBold mb="xs">{service.title}</TextBold>
+              {service.items.map((item, idx) => {
+                return (
+                  <ServiceCheckbox
+                    key={item.enum}
+                    services={services}
+                    setServices={setServices}
```

Comment:
perhaps can call it `setSelectedServices`?

Reply:
as I left state name "services" I will keep setServices